pkg/apis/storage/types.go

```diff
@@ -310,6 +327,36 @@ type CSIDriverSpec struct {
 	VolumeLifecycleModes []VolumeLifecycleMode
 }
 
+// FSGroupPolicy specifies if a CSI Driver supports modifying
+// volume ownership and permissions of the volume to be mounted.
+// More modes may be added in the future.
+type FSGroupPolicy string
+
+const (
+	// Heuristic indicates that each volume should be examined
+	// to determine if the volume ownership and permissions
+	// should be modified.
+	// This mode should be defined if it's expected that the
+	// fsGroup may need to be modified depending on the
+	// pod's SecurityPolicy.
+	HeuristicFSGroupPolicy FSGroupPolicy = "Heuristic"
+
+	// Supported indicates that CSI driver supports volume ownership
+	// and permission change via fsGroup, and Kubernetes may use fsGroup
+	// to change permissions and ownership of the volume to match user requested fsGroup in
+	// the pod's SecurityPolicy
+	// This mode should be defined if the fsGroup is expected
+	// to always change on mount.
+	SupportedFSGroupPolicy FSGroupPolicy = "Supported"
+
+	// Unsupported indicates that volumes will be mounted without performing
+	// any ownership or permission modifications, as the CSIDriver does not support
+	// these operations.
+	// This mode should be selected if the CSIDriver does not support
+	// fsGroup modifications, such as with the NFS driver.
```

Comment:
@jsafrane we were actually discussing that we should support fsgroup for nfs drivers?

Reply:
Some NFS servers can support fsGroup. It depends on the server setting - `root-squash` is used frequently. Kubelet is then not root and it can't change GID/permissions of random files on such a volume.

"This mode should be selected if the CSIDriver does not support fsGroup modifications, for example when Kubernetes cannot change ownership and permissions on a volume due to root-squash settings on a NFS volume.